app/controllers/dashboards_controller.rb

```diff
@@ -0,0 +1,62 @@
+class DashboardsController < ApplicationController
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+  layout 'dashboard'
+  before_action :authenticate_user!
+
+  def index
+    @page_title = 'Dashboard'
+  end
+
+  def new_user
+    authorize :dashboard, :new_user?
+    @page_title = 'Dashboard -> Add User'
+    @user = User.new({})
+  end
+
+  def create_user
+    authorize :dashboard, :create_user?
+    @user = User.new(user_params)
+    @organization = Organization.find(current_user.organization_id)
+    @user.organization = @organization
+    if @user.save
+      UserMailer.send_email_to_user(@user, params[:user][:password]).deliver_now
+      flash[:notice] = 'Added a user successfully'
+      redirect_to dashboard_show_user_path
+    else
+      render :new_user, status: :unprocessable_entity
+    end
+  end
+
+  def change_status
+    authorize :dashboard, :change_status?
+    @user = User.find_by(id: params[:id])
```

Comment:
as i can see, this code has used many times, can we dry this code?

Reply:
can we use Memoization if we are using this multiple time?

Reply:
Bhaiya, I did some R&D about it. Where each time parameter [:id] is different. For this reason, each time is returned by a different user, that is why I could not memorize it. One thing you can do is make another function that returns a user based on the parameters, but this could not be minimized.